odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx

```diff
@@ -0,0 +1,192 @@
+      name: collector?.name || '',
+      namespaceName: collector?.namespace?.name || '',
+      description: collector?.description || '',
+    }),
+    [collector]
+  );
+
+  const {
+    handleSubmit,
+    control,
+    reset,
+    setValue,
+    formState: { isValid },
+  } = useForm<CollectorFormDataValues>({
+    mode: 'all',
+    reValidateMode: 'onChange',
+    defaultValues: getDefaultValues(),
+  });
+
+  React.useEffect(() => {
+    reset(getDefaultValues());
+  }, [collector]);
+
+  const initialState = { error: '', isSuccessfulSubmit: false };
+  const [{ error, isSuccessfulSubmit }, setState] = React.useState<{
+    error: string;
+    isSuccessfulSubmit: boolean;
+  }>(initialState);
+
+  const clearState = () => {
+    setState(initialState);
+    reset();
+  };
+
+  const onSubmit = (data: CollectorFormDataValues) => {
+    const parsedData = { ...data };
+    (collector
+      ? updateCollector({
+          collectorId: collector.id,
+          collectorUpdateFormData: parsedData,
+        })
+      : registerCollector({ collectorFormData: parsedData })
+    ).then(
+      () => {
+        setState({ ...initialState, isSuccessfulSubmit: true });
+        clearState();
+      },
+      (response: Response) => {
+        setState({
+          ...initialState,
+          error: response.statusText || 'Unable to register datasource',
+        });
+      }
+    );
+  };
+
+  const formTitle = (
+    <Typography variant="h4" component="span">
+      {collector ? 'Edit ' : 'Add '}
+      Datasource
+    </Typography>
+  );
+
+  const formContent = () => (
+    <form id="datasource-create-form" onSubmit={handleSubmit(onSubmit)}>
+      <Typography variant="subtitle2" fontSize="0.73rem">
+        Fields with the <Asterisk>*</Asterisk> symbol are required to save
+        the Datasource
+      </Typography>
+      <Controller
+        name="name"
+        control={control}
+        rules={{
+          required: true,
+          validate: value => !!value.trim(),
+        }}
+        render={({ field }) => (
+          <AppTextField
+            {...field}
+            sx={{ mt: 1.5 }}
+            label="Name"
+            placeholder="e.g. Data Tower"
+            required
+            customEndAdornment={{
+              variant: 'clear',
+              showAdornment: !!field.value,
+              onCLick: () => field.onChange(''),
+              icon: <ClearIcon />,
+            }}
+          />
+        )}
+      />
+      <Controller
+        control={control}
+        name="namespaceName"
+        defaultValue={collector?.namespace?.name}
+        render={({ field }) => (
+          <NamespaceAutocompleteContainer controllerProps={field} />
+        )}
+      />
+      <Controller
+        name="description"
+        control={control}
+        render={({ field }) => (
+          <AppTextField
+            {...field}
+            sx={{ mt: 1.25 }}
+            label="Description"
+            placeholder="Datasource description"
```

Comment:
Please use the naming associated with the collector

Reply:
done